### ISSUE TITLE
Device Short Address not provided  in Group & Scene 0x8xxx messages family

### ISSUE THREAD

**Issue by KiwiHC16** [fairecasoimeme/ZiGate#123]
For example In message 0x80A6, the address of the device is not provided by the zigate.
Could we add this info in the message ?
And in all messages for Group and Scene.

Mandatory/Urgent:
View Scene Response / 0x80A0
Scene Membership Response / 0x80A6

Optional/Could wait:
Add Scene Response / 0x80A1
Remove Scene Response / 0x80A2
Remove All Scene Response / 0x80A3
Store Scene Response / 0x80A4

**Comment by KiwiHC16** [comment]
Very similar need as https://github.com/fairecasoimeme/ZiGate/pull/90/files

**Comment by fairecasoimeme** [comment]
OK but it's not so easy because the location where I have to add the feature concern a lot of command.
If I add src address between endpoint and clusterid (for example), we'll lost a lot of compatibility.
To save retro compatibility, we have to add srcAddress at the end of response. I don't know if it's clean.
what do you think ?

**Comment by KiwiHC16** [comment]
As soon as this is documented, we don't really care where it is.

In the heading it would be easier to read, to decode.

As some answers have variable length, it would complexify the decoding if it's at the end.

Or it could be at the end of the heading and before the variable part.

As this is a very important field and we are still in the ramp up phase of the solution I would say lets go for the hard now, it will ease a lot our live later.

Perhaps we should review all answer messages and secure the fact that we do it for all.

I think we should proceed if you know how to do it.

Perhaps @pipiche and @doubz could comment.

PS: Perhaps need to be carefull with potential impact on crc.

**Comment by pipiche38** [comment]
From my end, I think at the end is the best place, as we can differentiate from one version to the other based on the length of the payload.

**Comment by KiwiHC16** [comment]
Make sense, let's go for this solution.

**Comment by fairecasoimeme** [comment]
In aim to manage versioning, it's better at the end. The inconvenient is the reading of the code.
so, I'll add at the end
Thanks

**Comment by ISO-B** [comment]
Following issues are also somewhat related to this #116 #122 #39

**Comment by doudz** [comment]
There's a error in doc for response 0x8063, two src addr ;)

**Comment by pipiche38** [comment]
This is right. In fact It would be easier to document the data structure when in firmware 3.0d or in firmware 3.0f

**Comment by doudz** [comment]
Another mistake in response 0x80A7,  src addr is at the end
